### PR TITLE
Reduce Video Controller Complexity

### PR DESCRIPTION
**Pull request by lackolaxton**
Noticed when hosted in a k8s env it was holding onto memory. Cleaned up the video controller so i can dig into why it is doing so.

Looks to potentially be not disposing state/transcoding files but this will help me find out.

### CODE REVIEW

File: Jellyfin.Api/Helpers/StreamingHelper.cs [hunk 1]
```diff
@@ -25,33 +24,21 @@ namespace Jellyfin.Api.Helpers;
 /// <summary>
 /// The streaming helpers.
 /// </summary>
-public static class StreamingHelpers
+public class StreamingHelper(IMediaSourceManager mediaSourceManager, IUserManager userManager, ILibraryManager libraryManager, IServerConfigurationManager serverConfigurationManager, IMediaEncoder mediaEncoder, ITranscodeManager transcodeManager) : IStreamingHelper
```
Comment (Shadowghost) [inline Jellyfin.Api/Helpers/StreamingHelper.cs:27]: Usually we use a dedicated constructor

**Comment by nielsvanvelzen** [comment]
Please revert the API changes

**Comment by lackolaxton** [comment]
> Please revert the API changes

The actual API has not changed, only the auto API definition. I was hoping to bring the changes in because it has improved readability due to there only being 2 parameters rather than 10's.

**Comment by nielsvanvelzen** [comment]
The casing of the query parameters has changed.

**Comment by lackolaxton** [comment]
> The casing of the query parameters has changed.

Ill add the property name field to the model? I believe DOTNET should handle converting camelcase to Proper but leave it with me

File: Jellyfin.Api/Controllers/VideosController.cs [hunk 2]
```diff
@@ -253,225 +251,71 @@ public async Task<ActionResult> MergeVersions([FromQuery, Required, ModelBinder(
...
-        [FromQuery] string? liveStreamId,
-        [FromQuery] bool? enableMpegtsM2TsMode,
-        [FromQuery] [RegularExpression(EncodingHelper.ValidationRegex)] string? videoCodec,
-        [FromQuery] [RegularExpression(EncodingHelper.ValidationRegex)] string? subtitleCodec,
-        [FromQuery] string? transcodeReasons,
-        [FromQuery] int? audioStreamIndex,
-        [FromQuery] int? videoStreamIndex,
-        [FromQuery] EncodingContext? context,
-        [FromQuery] Dictionary<string, string> streamOptions)
+    public async Task<ActionResult> GetVideoStream([FromRoute][Required] Guid itemId, [FromQuery] VideoStreamRequest request)
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/VideosController.cs:261]: We moved away from using a single object for query parameters. Please revert this change.

**Review by crobibero — CHANGES_REQUESTED** [review]

File: Jellyfin.Api/Controllers/VideosController.cs (hunk 2 above)
Comment (lackolaxton) [reply Jellyfin.Api/Controllers/VideosController.cs]: Reverted.

File: Jellyfin.Api/Helpers/StreamingHelper.cs (hunk 1 above)
Comment (lackolaxton) [reply Jellyfin.Api/Helpers/StreamingHelper.cs]: Reverted.

**Comment by lackolaxton** [comment]
How do we get people to re review this?

**Comment by nielsvanvelzen** [comment]
Just wait. This PR is extremely low priority and our focus is on 10.9 related changes at the moment.

**Comment by lackolaxton** [comment]
Hi,
I have updated this.

Ideally I would like to get this in, as it will make it easier for me to investigate 2 issues i am seeing.

- High memory usage in K8s.
- Throttling transcoding is causing freezing
